lib/Parser/Parser.yy

```diff
@@ -0,0 +1,211 @@
+%{
+#include "Parser.hpp"
+#include "Basic/Tokens.hpp"
+#include <iostream>
+%}
+
+// Bison options
+%require "3.8"
+%language "c++"
+//%debug
+
+// Namespace for the generated parser
+%define api.namespace {glu}
+%define api.parser.class {BisonParser}
+%define api.token.constructor
+%define api.value.type variant
+%define parse.error verbose
+
+// Parameters for the scanner
+%parse-param { glu::Scanner& scanner }
+%lex-param { glu::Scanner& scanner }
+
+%code requires {
+    #include "Basic/SourceLocation.hpp"
+    namespace glu {
+        class Scanner;
+        class Token;
+    }
+}
+
+%code {
+    #include "Lexer/Scanner.hpp"
+    #include "Basic/Tokens.hpp"
+
+    static int tokenKindToInt(glu::TokenKind kind) {
+        return static_cast<int>(kind);
+    }
+
+
+    // Redefine yylex to call our scanner and return a symbol
+    #define yylex(scanner) ([&]() -> BisonParser::symbol_type { \
+        glu::Token tok = scanner.nextToken(); \
+        return BisonParser::symbol_type(tokenKindToInt(tok.getKind()), std::move(tok)); \
+    }())
+}
+
+// --- Explicit declaration of tokens with their values ---
+// The values used here are those defined in the TokenKind enum in the order of TokenKind.def
+%token <glu::Token> eof 0
+%token <glu::Token> ident 1
+
+%token <glu::Token> unterminatedBlockCommentError 2
+%token <glu::Token> unterminatedStringLitError 3
+%token <glu::Token> unknownCharError 4
+
+%token <glu::Token> structKw 5
```

Comment:
you can add string representations to keywords, separators and operators:
```suggestion
%token <glu::Token> structKw 5 "struct"
```
you can do that by defining those 3 respective macros in BisonTokens.cpp